GitUI/CommandsDialogs/SettingsDialog/SettingsTreeViewUserControl.cs

```diff
@@ -152,6 +152,10 @@ private void textBoxFind_TextChanged(object sender, EventArgs e)
             foreach (var node in _nodesFoundByTextBox)
             {
                 HighlightNode(node, true);
+                for (var parent = node.Parent; parent != null; parent = parent.Parent)
```

Comment:
I think I found a `do/while` loop more understandable for a such case (and keep the 'for' loop for case where the length is known at the beginning)

Reply:
The `for` loop has the big advantage that the loop variable is scope-limited.
I always use `for` if there's a loop variable and an iteration step for the loop variable.

Reply:
I personally would write it as `do..while` but for me this is not a deal breaker